app_feup/lib/view/Widgets/print_info_card.dart

```diff
@@ -47,6 +76,57 @@ class PrintInfoCard extends GenericCard {
     );
   }
 
+  /// Returns a widget with all the print balance movements.
+  Widget printBalanceMovementsWidget(movements, context) {
+    if (movements.length <= 0) {
+      return getEmptyContainer('Não existem movimentos recentes', context);
+    }
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: this.printBalanceMovements(context, movements),
+    );
+  }
+
+  /// Returns a list of widgets with movements to be displayed in the print card
+  List<Widget> printBalanceMovements(context, movements) {
+    final List<Widget> rows = movements.map<Widget>((movement) {
+      return this.balanceMovement(context, movement);
+    }).toList();
+
+    return rows.length <= 3 ? rows : rows.sublist(0, 3);
+  }
+
+  // Individual movement row
+  Widget balanceMovement(context, movement) {
+    return Container(
+      margin: EdgeInsets.only(top: 8),
+      child: RowContainer(
+        color: Theme.of(context).backgroundColor,
+        child: Container(
+          padding: EdgeInsets.all(11),
+          child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              mainAxisSize: MainAxisSize.max,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                Text(DateTime.parse(movement['datetime'])
+                    .toFormattedDateString()),
+                movement['value'][0] != '-'
```

Comment:
Can you extract this to a lambda, to clarify it is checking for a negative value?